src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -442,6 +472,20 @@ private static boolean hasLineBreakBefore(DetailAST rightCurly) {
     }
 
     /**
+     * Checks if right curly is alone with next token is SEMI on line.
+     * @param details for validation.
+     * @return true, if right curly has next token is SEMI and is alone whith SEMI on line.
+     */
+    private static boolean isAloneWithSemi(Details details) {
+        final DetailAST lcurly = details.lcurly;
+        final DetailAST rcurly = details.rcurly;
+        final DetailAST nextToken = details.nextToken;
+        return lcurly.getLineNo() != rcurly.getLineNo()
+            && nextToken.getType() == TokenTypes.SEMI
```

Comment:
please move "nextToken.getType() == TokenTypes.SEMI" to the first element of expression to make code easy to read.